Lib/http/server.py

```diff
@@ -282,7 +282,7 @@ def parse_request(self):
         requestline = str(self.raw_requestline, 'iso-8859-1')
         requestline = requestline.rstrip('\r\n')
         self.requestline = requestline
-        words = requestline.split()
+        words = requestline.rstrip(' ').split(' ')
```

Comment:
What's the reason for the `rstrip` here? (I understand `split()` does the equivalent of this but it also does `lstrip()`). Sure, if there's trailing spaces, we can just consider them part of the reason message?

Reply:
Removing the `.rstrip(' ')` would cause `test_version_none_get` to fail:
```
FAIL: test_version_none_get (Lib.test.test_httpservers.BaseHTTPServerTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../cpython/Lib/test/test_httpservers.py", line 167, in test_version_none_get
    self.assertEqual(res.status, HTTPStatus.NOT_IMPLEMENTED)
AssertionError: 400 != <HTTPStatus.NOT_IMPLEMENTED: 501>
```
(...since we get a blank version which trips the 400 down at line 306.)

> Sure, if there's trailing spaces, we can just consider them part of the path?

In general, they'd wind up tacked onto the version rather than the path, tripping the same 400.  We already (and appropriately, in my mind) respond 400 to requests like
```
b'GET /some/path  HTTP/1.0   '
```

It'd only end up on the path if the client was speaking HTTP/0.9 like 
```
b'GET /some/path   '
```

I could be talked into doing a `.strip(' ')` here instead; this would allow us to continue processing requests like
```
b'  GET /some/path HTTP/1.0   '
```
like we currently do (whereas my patch will cause an early exit with `Bad request syntax`).

However, considering that the bug was the result of lax parsing, I was inclined to do the minimal amount of requestline massaging to keep existing tests passing.